app/lib/forms/work_setting.rb

```diff
@@ -0,0 +1,51 @@
+module Forms
+  class WorkSetting < Base
+    VALID_WORK_SETTING_OPTIONS = %w[early_years_or_childcare a_school an_academy_trust a_16_to_19_educational_setting other].freeze
+
+    attr_accessor :work_setting
+
+    validates :work_setting, presence: true, inclusion: { in: VALID_WORK_SETTING_OPTIONS }
+
+    def self.permitted_params
+      %i[work_setting]
+    end
+
+    def after_save
+      # we are inferring `works_in_school` and `works_in_childcare` to maintain
+      # consistency with older records
```

Comment:
Should this after_save have a guard clause that exits if work_setting hasn't changed? Is it possible to detect that?

Reply:
I did have a look but couldn't see a tidy way of doing it, other than recording a list of modified fields inside `#save!`

Reply:
I think we'd have to add in [ActiveModel::Dirty](https://api.rubyonrails.org/classes/ActiveModel/Dirty.html) to get that functionality brought across from ActiveRecord. 

That'd be a lot of work though and not worth the time implementing.